File: server/src/catalog.ts
// Profil kataloğu. Uygulama bu kimliklerin çevirilerini kendi dil dosyalarında tutar;
// buraya kimlik eklenirse app/lib/l10n/*.arb dosyalarına da eklenmeli.

export const INTERESTS = [
  'coffee', 'travel', 'music', 'concerts', 'movies', 'series', 'books', 'photography',
  'art', 'cooking', 'foodie', 'wine', 'fitness', 'yoga', 'running', 'cycling',
  'hiking', 'camping', 'football', 'basketball', 'gaming', 'tech', 'fashion', 'dancing',
  'pets', 'nature', 'beach', 'meditation', 'anime', 'volunteering',
] as const;

export const PROMPTS = [
  'perfect_sunday', 'laugh', 'green_flag', 'travel_dream', 'unpopular_opinion',
  'simple_pleasures', 'looking_for', 'two_truths', 'first_date', 'song',
] as const;

export const LOOKING_FOR = ['relationship', 'casual', 'friendship', 'chat', 'unsure'] as const;
export const EDUCATION = ['high_school', 'bachelor', 'master', 'phd'] as const;
export const HABIT = ['no', 'sometimes', 'yes'] as const;
export const ZODIAC = [
  'aries', 'taurus', 'gemini', 'cancer', 'leo', 'virgo',
  'libra', 'scorpio', 'sagittarius', 'capricorn', 'aquarius', 'pisces',
] as const;

// Selfie doğrulamada istenen pozlar (uygulama kendi dilinde açıklamasını gösterir)
export const VERIFICATION_POSES = ['peace_sign', 'thumbs_up', 'hand_on_head', 'point_up', 'wave'] as const;

export const MAX_INTERESTS = 5;
export const MAX_PROMPTS = 3;

// Faz 16: kişisel profil vitrini. Renk (rozet/vurgu) ve arkaplan (kart zemini) ayrı seçilir; ''
// varsayılan marka görünümü demektir. Renkler/gradyanlar uygulamada core/catalog.dart'ta tutulur,
// burada sadece kimlikler doğrulanır (sunucu görsel bir şey saklamaz).
export const THEMES = ['coral', 'ocean', 'sunset', 'forest', 'lavender', 'rose'] as const;
export const CARD_BACKGROUNDS = ['default', 'ocean', 'sunset', 'forest', 'lavender', 'midnight'] as const;

// Faz 16: çizgi avatar. Renkler/çizim uygulamada core/catalog.dart'ta; burada sadece kimlik doğrulanır.
export const AVATAR_SKINS = ['light', 'medium', 'tan', 'brown', 'dark'] as const;
export const AVATAR_HAIR_STYLES = ['bald', 'short', 'long', 'curly'] as const;
export const AVATAR_HAIR_COLORS = ['black', 'brown', 'blonde', 'red', 'gray'] as const;
export const AVATAR_OUTFITS = ['coral', 'ocean', 'sunset', 'forest', 'lavender', 'midnight'] as const;
export const AVATAR_ACCESSORIES = ['none', 'glasses', 'hat', 'headphones'] as const;

// Faz 16: kendi oda. Statik yerleşim (gerçek zamanlı gezinme yok) — her eşya bir ızgara hücresinde.
export const ROOM_WALLPAPERS = ['plain', 'coral', 'ocean', 'sunset', 'forest', 'lavender'] as const;
export const ROOM_FLOORS = ['wood', 'tile', 'coral', 'ocean'] as const;
export const ROOM_ITEMS = ['sofa', 'bed', 'plant', 'lamp', 'tv', 'bookshelf', 'table', 'rug', 'window', 'picture'] as const;
export const ROOM_GRID_W = 4;
export const ROOM_GRID_H = 5;
export const ROOM_MAX_ITEMS = 12;

// Faz 16: günlük ruh hali. Serbest metin yok (moderasyon gerektirmez); 24 saatte kendiliğinden
// kaybolur (bkz. src/routes/profile.ts activeMood()). Emoji uygulamada core/catalog.dart'ta.
export const MOODS = [
  'happy', 'excited', 'relaxed', 'romantic', 'tired', 'stressed',
  'grateful', 'adventurous', 'lonely', 'busy', 'hopeful', 'bored',
] as const;
export const MOOD_TTL_MS = 24 * 60 * 60 * 1000;

File: app/lib/core/catalog.dart
// Profil kataloğu. Kimlikler sunucudaki server/src/catalog.ts ile aynı olmalı;
// çeviriler l10n dosyalarındaki select mesajlarında (interestLabel vb.).
import 'package:flutter/material.dart';

const interestEmoji = <String, String>{
  'coffee': '☕', 'travel': '✈️', 'music': '🎵', 'concerts': '🎤', 'movies': '🎬',
  'series': '📺', 'books': '📚', 'photography': '📷', 'art': '🎨', 'cooking': '🍳',
  'foodie': '🍜', 'wine': '🍷', 'fitness': '💪', 'yoga': '🧘', 'running': '🏃',
  'cycling': '🚴', 'hiking': '🥾', 'camping': '🏕️', 'football': '⚽', 'basketball': '🏀',
  'gaming': '🎮', 'tech': '💻', 'fashion': '👗', 'dancing': '💃', 'pets': '🐾',
  'nature': '🌿', 'beach': '🏖️', 'meditation': '🕯️', 'anime': '🍥', 'volunteering': '🤝',
};

List<String> get interestIds => interestEmoji.keys.toList();

const promptIds = [
  'perfect_sunday', 'laugh', 'green_flag', 'travel_dream', 'unpopular_opinion',
  'simple_pleasures', 'looking_for', 'two_truths', 'first_date', 'song',
];

const lookingForEmoji = <String, String>{
  'relationship': '💞',
  'casual': '🥂',
  'friendship': '👋',
  'chat': '💬',
  'unsure': '🤔',
};

const educationIds = ['high_school', 'bachelor', 'master', 'phd'];
const habitIds = ['no', 'sometimes', 'yes'];
const zodiacIds = [
  'aries', 'taurus', 'gemini', 'cancer', 'leo', 'virgo',
  'libra', 'scorpio', 'sagittarius', 'capricorn', 'aquarius', 'pisces',
];

const minInterests = 3;
const maxInterests = 5;
const maxPrompts = 3;
const promptMaxLength = 200;

// Faz 16: kişisel profil vitrini. '' = varsayılan marka rengi/zemini (uygulama genelindeki
// gradyanla aynı), o yüzden bu haritalarda ayrı bir '' girişi yok — boş kimlik varsayılanı işaretler.
const themeAccent = <String, Color>{
  'coral': Color(0xFFFF4D6D),
  'ocean': Color(0xFF0EA5E9),
  'sunset': Color(0xFFF97316),
  'forest': Color(0xFF22C55E),
  'lavender': Color(0xFFA78BFA),
  'rose': Color(0xFFEC4899),
};
List<String> get themeIds => themeAccent.keys.toList();

const cardBackgroundGradient = <String, List<Color>>{
  'default': [Color(0xFFFF4D6D), Color(0xFFFF8A5B)],
  'ocean': [Color(0xFF0EA5E9), Color(0xFF14B8A6)],
  'sunset': [Color(0xFFF97316), Color(0xFFEC4899)],
  'forest': [Color(0xFF16A34A), Color(0xFF14B8A6)],
  'lavender': [Color(0xFFA78BFA), Color(0xFFEC4899)],
  'midnight': [Color(0xFF1E293B), Color(0xFF4338CA)],
};
List<String> get cardBackgroundIds => cardBackgroundGradient.keys.toList();

Color themeColorOf(String id) => themeAccent[id] ?? themeAccent['coral']!;
List<Color> cardGradientOf(String id) => cardBackgroundGradient[id] ?? cardBackgroundGradient['default']!;

// Faz 16: çizgi avatar. Basit katmanlı şekillerle çizilir (varlık/görsel dosyası gerekmez).
const avatarSkinColor = <String, Color>{
  'light': Color(0xFFFFE0BD),
  'medium': Color(0xFFF1C27D),
  'tan': Color(0xFFE0AC69),
  'brown': Color(0xFFC68642),
  'dark': Color(0xFF8D5524),
};
List<String> get avatarSkinIds => avatarSkinColor.keys.toList();

const avatarHairStyles = ['bald', 'short', 'long', 'curly'];
const avatarHairColor = <String, Color>{
  'black': Color(0xFF2B2118),
  'brown': Color(0xFF6B4226),
  'blonde': Color(0xFFE8C275),
  'red': Color(0xFFB5502C),
  'gray': Color(0xFFB0B0B0),
};
List<String> get avatarHairColorIds => avatarHairColor.keys.toList();

// Kıyafet renkleri tema paletiyle aynı (tutarlı görünüm)
Color avatarOutfitColorOf(String id) => themeColorOf(id);
List<String> get avatarOutfitIds => themeIds;

const avatarAccessoryEmoji = <String, String>{'none': '', 'glasses': '👓', 'hat': '🎩', 'headphones': '🎧'};
List<String> get avatarAccessoryIds => avatarAccessoryEmoji.keys.toList();

// Faz 16: kendi oda. Statik yerleşim; her eşya sabit bir ızgara hücresine yerleştirilir.
const roomGridW = 4;
const roomGridH = 5;
const roomMaxItems = 12;

const roomWallpaperColor = <String, Color>{
  'plain': Color(0xFFF5EDE6),
  'coral': Color(0xFFFFE1E8),
  'ocean': Color(0xFFDCF3FA),
  'sunset': Color(0xFFFFE7D6),
  'forest': Color(0xFFE1F5EA),
  'lavender': Color(0xFFEEE6FB),
};
List<String> get roomWallpaperIds => roomWallpaperColor.keys.toList();
Color roomWallpaperColorOf(String id) => roomWallpaperColor[id] ?? roomWallpaperColor['plain']!;

const roomFloorColor = <String, Color>{
  'wood': Color(0xFFC8935B),
  'tile': Color(0xFFD9D2C7),
  'coral': Color(0xFFF6B8B8),
  'ocean': Color(0xFFA9D8E8),
};
List<String> get roomFloorIds => roomFloorColor.keys.toList();
Color roomFloorColorOf(String id) => roomFloorColor[id] ?? roomFloorColor['wood']!;

const roomItemEmoji = <String, String>{
  'sofa': '🛋️',
  'bed': '🛏️',
  'plant': '🪴',
  'lamp': '💡',
  'tv': '📺',
  'bookshelf': '📚',
  'table': '🪑',
  'rug': '🧺',
  'window': '🪟',
  'picture': '🖼️',
};
List<String> get roomItemIds => roomItemEmoji.keys.toList();
